src/neo-vm/Types/Integer.cs

```diff
@@ -8,11 +8,22 @@ namespace Neo.VM.Types
     [DebuggerDisplay("Type={GetType().Name}, Value={value}")]
     public class Integer : PrimitiveType
     {
-        private int _length = -1;
+        public const int MaxSize = 32;
+
+        private readonly int _length;
         private readonly BigInteger value;
 
         public Integer(BigInteger value)
         {
+            if (value.IsZero)
+            {
+                _length = 0;
+            }
+            else
+            {
+                _length = value.GetByteCount();
+                if (_length > MaxSize) throw new ArgumentException();
```

Comment:
We will check this twice sometimes, in ExecutionEngine and here

Reply:
We can remove the checks in `ExecutionEngine`. What do you think?

Reply:
I prefer to return false instead of trow exceptions. Maybe we can create a static method like
`bool TryCreate(..., out Integer v) `

Reply:
Believe me, this really won't be a bottleneck. The exception is only thrown once, and the contract fails. We should not waste time in such places.

Reply:
Then if we will check it here, we can remove it from `ExecutionEngine`.

Reply:
Done.